geo/src/algorithm/coordinate_position.rs

```diff
@@ -239,11 +239,6 @@ where
             return;
         }
 
-        // Ok to `unwrap` since we know `self` is non-empty, so bounding-rect is non-null
```

Comment:
Since bbox checks are typically pretty quick, this was a short-circuit for when the point is "far" outside the ring. 

I honestly expected to see regressions for that case, so I just now wrote an additional bench that tests points far from their ring:
https://github.com/georust/geo/commit/efac869b7b7430bb27bb91922bacdf4bcfaa1e21

...but I was wrong! You've sped up the winding number logic so much that even computing a bbox is slower.

Reply:
That seems well reasoned to me. Thank you for explaining.

Reply:
Just to make sure I'm understanding... this is the kind of case where we might expect a regression right?

Where the point (green) is horizontal (blue) to an inordinate number of edges in the polygon (red).

![image](https://user-images.githubusercontent.com/217057/230168573-af00a591-1a1a-4692-baeb-144c8efc4b62.png)

(forgive my graphic art skills)

Reply:
yes, exactly! we could prevent this with a bounding check, though not when the point is inside the bounds. for the latter some randomized ray direction shenanigans could help.